### DISCUSSION TITLE
I need help to share recoil root to other component and a way to get and set state outside of component

### DISCUSSION THREAD

**Discussion by Innovdata-damien** [facebookexperimental/Recoil#1839] [Q&A] [answered]
Hello ! I want to share recoilroot to an other react dom render using [Web components](https://fr.reactjs.org/docs/web-components.html). And i want to know a way to access to state from outside recoilroot like i did with redux  with creation store and storing it in variable. I don't know how to do it, if someone can help me with an example it's would be great

**Comment by drarmstr** [comment, marked as answer]
Recoil state is coupled with the particular version of state being rendered for the host `<RecoilRoot>` store, and isn't "global" per-se.  That said, from a component in that React component tree you can use something like `useRecoilCallback()` to create accessors and setters for reading and updating Recoil state from outside of rendering functions.  There are also third-party libraries like `recoil-nexus`, I think, that provide this pattern.

**Comment by Innovdata-damien** [comment]
Hello after a lot of search, i has seen some people using event emitter function with a lib like RxJS to update from outside a component is it a good way too ?
